### PR TITLE
upload images - Added two new overloads to the upload method  #5

### LINKED ISSUE #5: Add ability to upload images and videos
Currently, the Java SDK only allows for uploading documents. This means that when uploading images and videos, they are currently stored as VANTIQ Documents. 

This enhancement would enable the SDK to upload images and videos as VANTIQ Images/Videos, instead of only being able to upload them as documents.

### PR DESCRIPTION
**Pull request by namirfa**
Closes #5 

Added two new overloads to the upload method (one for async, one for sync). These overloads have a new parameter, `resourcePath`, which can be used to upload to `/resources/images` and `/resources/videos` respectively.

Added tests to make sure that the behavior works as expected for sample JPEG, PNG, and MP4 files.

I had to fix a bunch of the old tests in `VantiqIntegrationTests`, since they no longer worked, and had to modify the `TestType.json` document since it was out of date. I was unable to figure out how to get `testSubscribeType()` and `testSubscribeTopic()`, so for the time being I have ignored them. @steve-langley if you have any idea why those tests are failing, please let me know.

**TODO:**
- Add documentation for the new upload method
- Work with @steve-langley to handle the "magic" of getting this built and into maven.

### CODE REVIEW

**Comment by steve-langley** [comment]
I don't know much about the tests (I didn't write any of them) but I was able to get them running. When  I turned testSubscribeType and testSubscribeTopic back on they seemed to work fine for me. What errors were you seeing?

File: src/test/java/io/vantiq/client/VantiqConnectAuthTest.java [hunk 1]
```diff
@@ -168,7 +168,28 @@ public void testUnauthorizedSubscribe() throws Exception {
...
+            ResponseHandler nullResponseHandler = null; // Used to distinguish between different upload methods
+            vantiq.upload(new File("/foo/bar/file.txt"), "text/plain", "file.txt", nullResponseHandler);
+            fail("Should not allow unauthenticated requests");
+        } catch(IllegalStateException ex) {
+            assertThat(ex.getMessage(), CoreMatchers.is("Not authenticated"));
+        }
+    }
+
+    @Test
+    public void testUnauthroizedUploadImage() throws Exception {
```
Comment (steve-langley) [inline src/test/java/io/vantiq/client/VantiqConnectAuthTest.java]: This is mis-spelled - should be testUnauthorizedUploadImage.

File: src/test/java/io/vantiq/client/VantiqConnectAuthTest.java [hunk 2]
```diff
@@ -168,7 +168,28 @@ public void testUnauthorizedSubscribe() throws Exception {
...
+        try {
+            vantiq.upload(new File("/foo/bar/file.jpg"), "image/jpeg", "file.jpg", "/resources/images", null);
+            fail("Should not allow unauthenticated requests");
+        } catch(IllegalStateException ex) {
+            assertThat(ex.getMessage(), CoreMatchers.is("Not authenticated"));
+        }
+    }
+
+    @Test
+    public void testUnauthroizedUploadVideo() throws Exception {
```
Comment (steve-langley) [inline src/test/java/io/vantiq/client/VantiqConnectAuthTest.java]: This is mis-spelled - should be testUnauthorizedUploadVideo.

**Comment by namirfa** [comment]
@steve-langley 
Thanks for the review, I fixed the typos that you pointed out. As for the `VantiqIntegrationTests`, those two tests both fail for me on this line `assertThat("Connected", callback.isConnected(), is(true));`

Not sure if it's a configuration mistake on my end or what, but I cannot get the connection to succeed.

**Comment by steve-langley** [comment]
Yeah, I don't really understand these tests either :)

Maybe it's a timing error; you could try changing UnitTestSubscriptionCallback.java at line 74 - instead of 2000 try 5000. Maybe it's taking longer that 2 seconds to connect and a longer wait would fix it. (Not sure why that would be.)

**Comment by namirfa** [comment]
Thanks again for helping out. I tried that earlier and again just now, but it still doesn't work!

When you ran the tests, did you set up your run configuration to connect to a localhost?

**Comment by steve-langley** [comment]
Yes, I pointed at my server running on localhost. I had also imported all the resources from src/test/resources/intgTest. That seems to be necessary to make all the other tests work, but I don't see how it matters for Topics. It would matter for Types. I assume you have "TestType" defined or the other tests would have failed too.

**Comment by namirfa** [comment]
Yep I did the exact same things to set up my server running on localhost, but for some reason they keep failing! Maybe we can compare when I am back in the office, hopefully tomorrow.

**Comment by namirfa** [comment]
@steve-langley 
I updated the documentation to reflect the new upload changes. If that looks good, let me know when you have time to handle the "publishing" step.

**Comment by fhcarter** [comment]
We probably don't need to publish until we're about ready to release, correct?

**Pull request merged**